### PR TITLE
feat(otel): GA for cmake

### PR DESCRIPTION
**Pull request by dbolduc**
Part of the work for #12726 

I think we need to wait until our vcpkg port has been updated before changing that part of the docs: https://github.com/googleapis/google-cloud-cpp/blob/f251ffce0ad0ddd96c87d8e4aa00907ec9403a91/google/cloud/opentelemetry/quickstart/README.md?plain=1#L168

- - -
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/12735)

### CODE REVIEW

File: cmake/GoogleCloudCppFeatures.cmake [hunk 1]
```diff
@@ -357,7 +365,7 @@ macro (google_cloud_cpp_enable_cleanup)
         OR (compute IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
         OR (compute_features)
         OR (experimental-bigquery_rest IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
-        OR (experimental-opentelemetry IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
+        OR (opentelemetry IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
```
Comment (coryan) [inline cmake/GoogleCloudCppFeatures.cmake:368]: If you are still supporting (with a warning) `experimental-opentelemetry` you need to keep it here too.

File: cmake/GoogleCloudCppFeatures.cmake [hunk 2]
```diff
@@ -322,6 +322,14 @@ macro (google_cloud_cpp_enable_deps)
...
     endif ()
     if (experimental-opentelemetry IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
+        message(
+            WARNING
+                "The feature `experimental-opentelemetry` is deprecated. Please use `opentelemetry` instead."
+        )
+        list(REMOVE_ITEM GOOGLE_CLOUD_CPP_ENABLE experimental-opentelemetry)
+        list(INSERT GOOGLE_CLOUD_CPP_ENABLE 0 opentelemetry)
+    endif ()
+    if (opentelemetry IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
```
Comment (coryan) [inline cmake/GoogleCloudCppFeatures.cmake:332]: Should we include `opentelemetry` in `GOOGLE_CLOUD_CPP_GA_LIBRARIES`?  I think we should.

File: google/cloud/google_cloud_cpp_common.cmake [hunk 3]
```diff
@@ -171,7 +171,7 @@ target_link_libraries(
            Threads::Threads
            OpenSSL::Crypto)
 
-if (experimental-opentelemetry IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
+if (opentelemetry IN_LIST GOOGLE_CLOUD_CPP_ENABLE)
```
Comment (coryan) [inline google/cloud/google_cloud_cpp_common.cmake:174]: If you still support `experimental-opentelemetry`, then you cannot remove it from here either.

File: ci/verify_quickstart/CMakeLists.txt [hunk 4]
```diff
@@ -105,7 +105,7 @@ foreach (feature IN LISTS FEATURES)
         add_make_quickstart_target("${feature}" storage quickstart_grpc)
         continue()
     endif ()
-    if ("${feature}" STREQUAL "experimental-opentelemetry")
+    if ("${feature}" STREQUAL "opentelemetry")
```
Comment (coryan) [inline ci/verify_quickstart/CMakeLists.txt:108]: Seems like the code around line 126 would work fine now?

**Review by coryan — CHANGES_REQUESTED** [review]

File: cmake/GoogleCloudCppFeatures.cmake (hunk 1 above)
Comment (dbolduc) [reply cmake/GoogleCloudCppFeatures.cmake:368]: If we find `experimental-opentelemetry` in the list, we replace it with `opentelemetry`.

https://github.com/googleapis/google-cloud-cpp/blob/269a29dfb4b8131d4fa0da42cf1817df49cb7d2f/cmake/GoogleCloudCppFeatures.cmake#L324-L331

File: ci/verify_quickstart/CMakeLists.txt (hunk 4 above)
Comment (dbolduc) [reply ci/verify_quickstart/CMakeLists.txt:108]: We can't call `add_make_quickstart_target()` yet.

File: google/cloud/google_cloud_cpp_common.cmake (hunk 3 above)
Comment (dbolduc) [reply google/cloud/google_cloud_cpp_common.cmake:174]: See above. We replace `experimental-opentelemetry` with `opentelemetry`.

**Review by coryan — APPROVED** [review]
Feel free to merge as-is while you experiment with a better alternative.

**Pull request merged**